parlai/core/worlds.py

```diff
@@ -150,6 +150,12 @@ def share(self):
         shared_data['agents'] = self._share_agents()
         return shared_data
 
+    def clone(self):
+        """
+        Create a duplicate of the world.
+        """
+        return type(self)(opt=copy.deepcopy(self.opt), agents=None, shared=self.share())
```

Comment:
It's the safest thing to do, so that if one world modifies anything in the opt it doesn't modify the content of the other world's opt. Though it costs memory to do so. I'll leave it up to you all who have more context on the tradeoffs that are best for this situation!

Reply:
`world.__init__` actually call deepcopy. But we do that for agents too. No guarantee a subchild behaves properly.

Reply:
Okay, in that case I'll keep the `deepcopy` here. If the `world.__init__` handles `deepcopy` that's great, and this intermediary copy will be garbage collected. If it doesn't, the `deepcopy` here will provide some extra safety.